### PR TITLE
/api/create-agent Implementation (GO->dart)

### PR DESCRIPTION
**Pull request by keyonghan**
This PR implements /api/create-agent, and creates corresponding tests.

### CODE REVIEW

File: app_dart/lib/src/request_handlers/create_agent.dart [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    final List<String> capabilities =
+        requestData[capabilitiesParam].cast<String>().toList();
+    final DatastoreService datastore = datastoreProvider();
+    final AgentService agentService = agentServiceProvider();
+    final Key key = datastore.db.emptyKey.append(Agent, id: agentId);
+
+    Agent agent;
+    AgentAuthToken agentAuthToken;
+
+    if (await datastore.db.lookupValue<Agent>(key, orElse: () => null) ==
```
Comment (tvolkert) [inline app_dart/lib/src/request_handlers/create_agent.dart]: Nit: if you structured this as `if (...) { throw ... }`, then you could set the `agent` and `agentAuthToken` variables when you declare them.

File: app_dart/lib/src/request_handlers/create_agent.dart [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        healthCheckTimestamp: DateTime.now().millisecondsSinceEpoch,
+        authToken: agentAuthToken.hash,
+        isHealthy: false,
+        key: key,
+      );
+    } else {
+      throw BadRequestException('Agent ID: $agentId already exists');
+    }
+
+    if (agent == null) {
```
Comment (tvolkert) [inline app_dart/lib/src/request_handlers/create_agent.dart]: ... and then you can get rid of this null check

File: app_dart/lib/src/request_handlers/create_agent.dart (hunk 1 above)
Comment (keyonghan) [reply app_dart/lib/src/request_handlers/create_agent.dart]: You are right! Just updated accordingly.

File: app_dart/lib/src/request_handlers/create_agent.dart (hunk 2 above)
Comment (keyonghan) [reply app_dart/lib/src/request_handlers/create_agent.dart]: Yes. Removed.

**Pull request merged**